lab-abigail/server.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const Client = require('./model/client.js');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = module.exports = net.createServer();
+const PORT = process.env.PORT || 3000;
+
+server.listen(PORT, function() {
+  console.log(`Listening on: ${PORT}`);
+});
+
+const pool = [];
+
+ee.on('default', function(client, string) {
+  client.socket.write(`Not a valid command: ${string.split(' ', 1)}\n`);
+});
+
+ee.on('/all', function(client, string) {
+  pool.forEach(c => c.socket.write(`${client.nickName}: ${string}\n`));
+});
+
+ee.on('/nick', function(client, string) {
+  client.original = client.nickName;
+  client.nickName = string.trim().split(' ').join('').toUpperCase();
+  client.socket.write(`You changed your nickname to from ${client.original} to ${client.nickName}\n`);
+  pool.forEach(pool => {
+    pool.socket.write(`${client.original} changed his/her name to ${client.nickName}\n`);
+  });
+});
+
+ee.on('/dm', function(client, string) {
+  let nickName = string.split(' ').shift().trim().toUpperCase();
+  let message = string.split(' ').slice(1).join(' ').trim();
+
+  pool.forEach(dm => {
+    if (dm.nickName === nickName) {
+      dm.socket.write(`Private Message from ${client.nickName}: ${message}\n`);
+    }
+  });
+  client.socket.write(`You private messaged ${nickName}: ${message}\n`);
+});
+
+server.on('connection', socket => {
+  let client = new Client(socket);
+  pool.push(client);
+  pool.forEach(c => c.socket.write(`${client.nickName} has connected!\n`));
+
+  socket.on('data', data => {
+    let command = data.toString().split(' ').shift().trim();
+    // console.log(`${client.nickName}: ${data}`);
+    if(command.startsWith('/')) {
+      ee.emit(command, client, data.toString().split(' ').slice(1).join(' '));
+      return;
+    }
+    ee.emit('default', client, data.toString());
+  });
+
+
+  socket.on('error', error => {
+    console.log(`Error: ${error}`);
+  });
+
+  socket.on('close', data => {
+    ee.emit('quit', client, data.toString());
```

Comment:
No points lost for this, but Line 66 actually doesn't do anything. In order for this do something substantial, you would need an additional ``ee.on('quit, some callback function...)`` that is placed at the top of this module along with the other ``ee.on`` events. 

Compare this with Line 57 of this module where you have ``    ee.emit('default', client, data.toString());``. Further up on Line 16, you have the listener for ``default`` that will actually trigger something when the word ``default`` is detected. 

I hope this makes sense, but if not, ask!